### PR TITLE
Capistrano 3 and 2 updates

### PR DESCRIPTION
**Pull request by dprvig**
Very similar to https://github.com/Instrumental/instrumental_agent-ruby/pull/61. The main difference here is that I reformatted the Capistrano 2 recipe to look more like the recipe from Seuros' gems. That update has no functional changes, but I think it looks nice. Also includes the Cap 3 recipe. Feel free to close this one if the cap 2 updates aren't desired.

### CODE REVIEW

**Comment by dprvig** [comment]
Realized I didn't link the Seuros stuff: https://github.com/seuros/capistrano-sidekiq/blob/master/lib/capistrano/tasks/capistrano2.rb

**Comment by janxious** [comment]
Could I talk you into cherry-picking the commits here without the versioning and changelog stuff? We do versioning and changelog changes as part of the release instead of as interim commits.

**Comment by dprvig** [comment]
@janxious Sure thing! I've gone ahead and removed the versioning and changelong stuff :)

**Comment by dprvig** [comment]
Not exactly sure why https://travis-ci.org/Instrumental/instrumental_agent-ruby/jobs/308037293 failed. Looks like a segfault when trying to run `bundle`.

**Comment by janxious** [comment]
Intermittent, seemingly. Re-ran and everything works

**Comment by mediocretes** [comment]
I think I prefer this to #61, but either is fine with me.

**Comment by dprvig** [comment]
I prefer this branch as well. Thanks for reviewing! With that said, is there any kind of ETA on this getting merged? Currently pointing to my own fork for local development :)

**Comment by janxious** [comment]
We haven't tested the capistrano 2 path still functions as expected, so whenever we drop it in a project of that sort and test it. Probably this week some time.

File: lib/instrumental/capistrano/capistrano2.rb [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+    end
+
+    desc "send a notice to instrumental about the deploy"
+    task :record_deploy_notice do
+      start_at                   = fetch(:instrumental_deploy_start, Time.now)
+      end_at                     = fetch(:instrumental_deploy_end, start_at)
+      deploy_duration_in_seconds = end_at - start_at
+      deployer                   = fetch(:deployer)
+      agent_options              = { :synchronous => true }
+      agent_options[:collector]  = instrumental_host if fetch(:instrumental_host)
```
Comment (janxious) [inline lib/instrumental/capistrano/capistrano2.rb]: This needs to be `fetch(:instrumental_host, false)` to allow for the value to be not set. This uses `Hash#fetch`,  which raises an exception in the case a given key is not in a hash. Providing a fallback of `false` allows this to fail gracefully while maintaining the patterns introduced here.

**Comment by dprvig** [comment]
@janxious Updated!

File: lib/instrumental/capistrano/capistrano2.rb [hunk 2]
```diff
@@ -34,7 +34,7 @@
       deploy_duration_in_seconds = end_at - start_at
       deployer                   = fetch(:deployer)
       agent_options              = { :synchronous => true }
-      agent_options[:collector]  = instrumental_host if fetch(:instrumental_host)
+      agent_options[:collector]  = fetch(:instrumental_host, false)
```
Comment (janxious) [inline lib/instrumental/capistrano/capistrano2.rb]: Sorry, needs to be `instrumental_host if fetch(:instrumental_host, false)`. I could have been more clear in my other message.

Comment (dprvig) [reply lib/instrumental/capistrano/capistrano2.rb]: Ah, wasn't sure. Easy enough tho.

**Comment by janxious** [comment]
Tested this and everything worked as expected in cap 2 world.

**Comment by janxious** [comment]
https://github.com/Instrumental/instrumental_agent-ruby/tree/v2.1.0 includes these changes as well as the safe instrumental host lookup for cap 3 users.

**Pull request merged**